src/odemis/gui/comp/overlay/base.py

```diff
@@ -313,9 +313,8 @@ def __init__(self, cnvs, label=None):
             self.add_label(label)
 
         # When an overlay is active, it will process mouse events
-        # So, check for this attribute if the sub class needs to process an event only if it's
-        # active.
-        self.active = False
+        # So, check for this VA if the sub class needs to process an event only if it's active.
+        self.active = BooleanVA(False, setter=self._on_active_va)
```

Comment:
Instead of using the setter, use a subscription. The difference is that the setter is called before the value of the VA is changed. It can decide what should be the actual value set, or raise an error. Here, it's better to call the (de)activate() function after the value has changed.
```python 
self.active = BooleanVA(False)
self.active.subscribe(self._on_active_va)
```